nemo/collections/nlp/models/text_normalization_as_tagging/thutmose_tagger.py

```diff
@@ -0,0 +1,330 @@
+from typing import Dict, List, Optional
+
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+
+from nemo.collections.common.losses import CrossEntropyLoss
+from nemo.collections.nlp.data.text_normalization_as_tagging import (
+    ThutmoseTaggerDataset,
+    ThutmoseTaggerTestDataset,
+    bert_example,
+    tagging,
+)
+from nemo.collections.nlp.data.text_normalization_as_tagging.utils import read_label_map, read_semiotic_classes
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
+from nemo.collections.nlp.models.nlp_model import NLPModel
+from nemo.collections.nlp.modules.common.token_classifier import TokenClassifier
+from nemo.collections.nlp.parts.utils_funcs import tensor2list
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+from nemo.core.neural_types import ChannelType, LogitsType, MaskType, NeuralType
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+__all__ = ['ThutmoseTaggerModel']
+
+
+@experimental
+class ThutmoseTaggerModel(NLPModel):
+    """
+    BERT-based tagging model for ITN, inspired by LaserTagger approach.
+    It maps spoken-domain input words to tags:
+        KEEP, DELETE, or any of predefined replacement tags which correspond to a written-domain fragment.
+    Example: one hundred thirty four -> _1 <DELETE> 3 4_
+    """
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "input_ids": NeuralType(('B', 'T'), ChannelType()),
+            "input_mask": NeuralType(('B', 'T'), MaskType()),
+            "segment_ids": NeuralType(('B', 'T'), MaskType()),
+            # "attention_mask": NeuralType(('B', 'T'), MaskType(), optional=True),
+        }
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {"logits": NeuralType(('B', 'T', 'D'), LogitsType())}
+
+    @property
+    def input_module(self):
+        return self
+
+    @property
+    def output_module(self):
+        return self
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None) -> None:
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        label_map_file = self.register_artifact("label_map", cfg.label_map)
+        semiotic_classes_file = self.register_artifact("semiotic_classes", cfg.semiotic_classes)
+        self.label_map = read_label_map(label_map_file)
+        self.semiotic_classes = read_semiotic_classes(semiotic_classes_file)
+
+        self.num_labels = len(self.label_map)
+        self.id_2_tag = {tag_id: tagging.Tag(tag) for tag, tag_id in self.label_map.items()}
+        self.max_sequence_len = cfg.get('max_sequence_len', self.tokenizer.tokenizer.model_max_length)
+
+        # setup to track metrics
+        # we will have (len(self.semiotic_classes) + 1) labels, last one stands for WRONG
+        label_ids = self.semiotic_classes.copy()
+        label_ids["WRONG"] = len(self.semiotic_classes)
```

Comment:
what does `WRONG` mean? how about `SELF`?

Reply:
`SELF` is `PLAIN` in terms of semiotic classes